compositor_pipeline/src/pipeline/input/mp4.rs

```diff
@@ -0,0 +1,269 @@
+}
+
+impl Mp4 {
+    pub fn new(options: Mp4Options) -> Result<(Self, ChunksReceiver), Mp4Error> {
+        let (input_path, delete_file) = match options.source {
+            Source::Url(url) => {
+                let mut file_response = reqwest::blocking::get(url)?;
+
+                let mut path = std::env::var("LIVE_COMPOSITOR_TMP_PATH")
+                    .map(PathBuf::from)
+                    .unwrap_or(std::env::temp_dir());
+
+                path.push(format!(
+                    "live-compositor-user-file-{}.mp4",
+                    rand::random::<u64>()
+                ));
+
+                let mut file = std::fs::File::create(&path)?;
+
+                std::io::copy(&mut file_response, &mut file)?;
+
+                (path, true)
+            }
+            Source::File(path) => (path, false),
+        };
+
+        let input_file = std::fs::File::open(input_path.clone())?;
+        let size = input_file.metadata()?.size();
+        let reader = mp4::Mp4Reader::read_header(input_file, size)?;
+        let stop_video_reader = Arc::new(AtomicBool::new(false));
+
+        let (video_thread, video_receiver) =
+            spawn_video_reader(reader, options.input_id.clone(), stop_video_reader.clone())?;
+
+        Ok((
+            Self {
+                input_id: options.input_id,
+                video_thread: Some(video_thread),
+                delete_file,
+                path: input_path,
+                stop_video_reader,
+            },
+            ChunksReceiver {
+                video: Some(video_receiver),
+                audio: None,
+            },
+        ))
+    }
+}
+
+impl Drop for Mp4 {
+    fn drop(&mut self) {
+        if let Some(thread) = self.video_thread.take() {
+            self.stop_video_reader
+                .store(true, std::sync::atomic::Ordering::Relaxed);
+            thread.join().unwrap();
+        }
+
+        if self.delete_file {
+            std::fs::remove_file(&self.path).unwrap();
+        }
+    }
+}
+
+fn spawn_video_reader(
+    mut reader: Mp4Reader<File>,
+    input_id: InputId,
+    stop_thread: Arc<AtomicBool>,
+) -> Result<(JoinHandle<()>, Receiver<EncodedChunk>), Mp4Error> {
+    let (&track_id, track, avc1) = reader
+        .tracks()
+        .iter()
+        .find_map(|(id, track)| {
+            let track_type = match track.track_type() {
+                Ok(t) => t,
+                Err(_) => return None,
+            };
+
+            let media_type = match track.media_type() {
+                Ok(t) => t,
+                Err(_) => return None,
+            };
+
+            if track_type != mp4::TrackType::Video
+                || media_type != mp4::MediaType::H264
+                || track.trak.mdia.minf.stbl.stsd.avc1.is_none()
+            {
+                return None;
+            }
+
+            track
+                .trak
+                .mdia
+                .minf
+                .stbl
+                .stsd
+                .avc1
```

Comment:
![unga](https://github.com/membraneframework/video_compositor/assets/9753141/822d7032-0b33-4b64-a370-69ee7d3aaa82)

Reply:
You haven't seen `length_minus_one` yet :P

Reply:
@WojciechBarczynski it's `size_length_minus_one` actually :))